kai/reactive_codeplanner/agent/analyzer_fix/agent.py

```diff
@@ -0,0 +1,195 @@
+
+# Input information
+
+## Input File
+
+File name: "{{ src_file_name }}"
+Source file contents:
+```{{ src_file_language }}
+{{ src_file_contents | safe }}
+```
+
+## Issues
+
+{% for incident in incidents %}
+### incident {{ loop.index0 }}
+incident to fix: "{{ incident.message | safe }}"
+Line number: {{ incident.line_number }}
+{% if incident.solution_str is defined %}
+{{ incident.solution_str | safe }}
+{% endif %}
+{% endfor %}
+
+# Output Instructions
+Structure your output in Markdown format such as:
+
+## Reasoning
+Write the step by step reasoning in this markdown section. If you are unsure of a step or reasoning, clearly state you are unsure and why.
+
+## Updated Java File
+```java
+// Write the updated file for Quarkus in this section. If the file should be removed, make the content of the updated file a comment explaining it should be removed.
+```
+
+## Additional Information (optional)
+
+If you have any additional details or steps that need to be performed, put it here.
+
+    """
+    )
+
+    def __init__(
+        self,
+        model_provider: ModelProvider,
+        retries: int = 1,
+    ) -> None:
+        self._model_provider = model_provider
+        self._retries = retries
+
+    def execute(self, ask: AgentRequest) -> AnalyzerFixResponse:
+
+        if not isinstance(ask, AnalyzerFixRequest):
+            return AnalyzerFixResponse(
+                encountered_errors=["invalid request type"],
+                file_to_modify=None,
+                reasoning=None,
+                additional_information=None,
+                updated_file_content=None,
+            )
+
+        file_name = os.path.basename(ask.file_path)
+
+        language = guess_language(ask.file_content, file_name)
+
+        content = self.chat_message_template.render(
+            src_file_contents=ask.file_content,
+            src_file_name=file_name,
+            src_file_language=language,
+            incidents=ask.incidents,
+        )
+
+        aimessage = self._model_provider.invoke(
+            [self.system_message, HumanMessage(content=content)]
+        )
+
+        resp = self.parse_llm_response(aimessage)
+        return AnalyzerFixResponse(
+            encountered_errors=[],
+            file_to_modify=ask.file_path,
+            reasoning=resp.reasoning,
+            additional_information=resp.addional_information,
+            updated_file_content=resp.java_file,
+        )
+
+    def parse_llm_response(self, message: BaseMessage) -> _llm_response:
+        """Private method that will be used to parse the contents and get the results"""
+
+        lines_of_output = cast(str, message.content).splitlines()
+
+        in_java_file = False
+        in_reasoning = False
+        in_additional_details = False
+        java_file = ""
+        reasoning = ""
+        additional_details = ""
+        for line in lines_of_output:
+            if line.strip() == "## Reasoning":
```

Comment:
I think we need to make this more error-proof in future. Strict matching on section headers sometimes fails when LLMs don't quite follow the exact syntax

Reply:
Yeah, I agree. would a followup issue on a more comprehensive LLM response parsing be a good idea?

Reply:
sounds good to me